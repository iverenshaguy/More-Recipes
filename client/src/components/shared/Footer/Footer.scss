@import '../../../assets/scss/_modules.scss';

.footer {
  position: absolute;
  z-index: -5;
  /* bottom: 0; */
  width: 100%;
  line-height: 2rem;
  padding-top: 1rem;
  background-color: transparent;
  color: white !important;
  text-align: center;

  @at-root .login-signup-body & {
    line-height: 1rem;
  }

  @at-root .search-recipes &,
    .view-recipe &,
    .add-recipe &,
    .favorite-recipes & {
    position: relative;
  }

  @at-root .user-recipe & {
    line-height: 0.5rem;
  }

  @at-root .user-profile & {
    position: relative;
    line-height: 2rem;
    color: $darkgrey !important;
  }

  @media only screen and (max-width: 500px) {
    & {
      position: relative;
    }
  }
}
