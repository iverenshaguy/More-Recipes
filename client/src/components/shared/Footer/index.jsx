import React from 'react';
import './Footer.scss';

/**
 * @exports
 * @function Footer
 * @extends Component
 * @returns {JSX} Footer
 */
const Footer = () => (
  <footer className="footer">
    <div className="container">
      <p>Â© {(new Date()).getFullYear()} More Recipes</p>
    </div>
  </footer>
);

export default Footer;
