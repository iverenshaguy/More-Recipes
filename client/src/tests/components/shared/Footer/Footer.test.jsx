import React from 'react';
import Footer from '../../../../components/shared/Footer';

const setup = () => {
  const mountedWrapper = mount(<Footer />);
  const shallowWrapper = shallow(<Footer />);

  return {
    mountedWrapper, shallowWrapper
  };
};

describe('Footer', () => {
  it('renders correctly', () => {
    const { shallowWrapper } = setup();
    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('renders correctly when fully mounted', () => {
    const { mountedWrapper } = setup();
    expect(toJson(mountedWrapper)).toMatchSnapshot();
    mountedWrapper.unmount();
  });
});
