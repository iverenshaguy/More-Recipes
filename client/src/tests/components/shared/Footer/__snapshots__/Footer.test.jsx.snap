// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer
  className="footer"
>
  <div
    className="container"
  >
    <p>
      © 
      2018
       More Recipes
    </p>
  </div>
</footer>
`;

exports[`Footer renders correctly when fully mounted 1`] = `
<Footer>
  <footer
    className="footer"
  >
    <div
      className="container"
    >
      <p>
        © 
        2018
         More Recipes
      </p>
    </div>
  </footer>
</Footer>
`;
